src/main/java/nl/tudelft/lifetiles/traverser/models/EmptySegmentTraverser.java

```diff
@@ -0,0 +1,186 @@
+package nl.tudelft.lifetiles.traverser.models;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+
+import nl.tudelft.lifetiles.graph.models.Edge;
+import nl.tudelft.lifetiles.graph.models.Graph;
+import nl.tudelft.lifetiles.graph.models.sequence.SegmentEmpty;
+import nl.tudelft.lifetiles.graph.models.sequence.Sequence;
+import nl.tudelft.lifetiles.graph.models.sequence.SequenceSegment;
+
+/**
+ * Places empty vertices in the graph where there are unified gaps.
+ *
+ * @author Jos
+ *
+ */
+public class EmptySegmentTraverser {
+    /**
+     * Traverser's graph.
+     */
+    private Graph<SequenceSegment> graphVar;
+
+    /**
+     * Traverses the graph.
+     *
+     * @param graph
+     *            the graph to traverse.
+     * @return the traversed of the graph.
+     */
+    public final Graph<SequenceSegment> traverseGraph(
+            final Graph<SequenceSegment> graph) {
+        graphVar = graph;
+        traverseGraph();
+        return this.graphVar;
+    }
+
+    /**
+     * Traverses the graph.
+     */
+    public final void traverseGraph() {
```

Comment:
Should rename this to something more descriptive. 'traverse' does not tell me what it will do. Neither is it explained in the corresponding javadoc.